circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -531,6 +558,68 @@ private enum class InstantiationType {
   CLASS
 }
 
+private enum class CodegenMode {
+  ANVIL {
+    override fun annotateFactory(builder: TypeSpec.Builder, scope: TypeName) {
+      builder.addAnnotation(
+        AnnotationSpec.builder(ContributesMultibinding::class).addMember("%T::class", scope).build()
+      )
+    }
+  },
+  HILT {
```

Comment:
For posterity, can we add kdocs to both of these that include sample snippets of the generated code/annotations? Makes it easier to see at a glance what it's going to do